### PR TITLE
:bug: e2e: PrivateKcpServer enable audit-policy

### PR DESCRIPTION
**Pull request by hardys**
## Summary

Otherwise we end up with zero-length audit logs in CI...

For example see [this](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/kcp-dev_kcp/2594/pull-ci-kcp-dev-kcp-main-e2e/1613198333013135360/artifacts/e2e/e2e/artifacts/TestUserHomeWorkspaces/Create_a_workspace_in_the_non-existing_home_and_have_it_created_automatically_through_~/3662491291/artifacts/kcp/main/) - AFAICS all tests using PrivateKcpServer have this problem because we only set the audit-policy flag in the [SharedKcpServer fixture](https://github.com/kcp-dev/kcp/blob/main/test/e2e/framework/kcp.go#L151)

### CODE REVIEW

**Comment by hardys** [comment]
/hold 

This requires further changes since some tests like `TestAuditLogs` set the flag explicitly

**Review by stevekuznetsov — APPROVED** [review]

**Comment by stevekuznetsov** [comment]
LGTM once you update the other tests.

**Comment by hardys** [comment]
> LGTM once you update the other tests.

Thanks @stevekuznetsov - I decided to just default the flag so we only set when not explicitly passed by the test - we can see that working e.g [here](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/kcp-dev_kcp/2591/pull-ci-kcp-dev-kcp-main-e2e/1613197205659717632/artifacts/e2e/e2e/artifacts/TestUserHomeWorkspaces/Create_a_workspace_in_the_non-existing_home_and_have_it_created_automatically_through_~/3272441777/artifacts/kcp/main/)

We also need to de-duplicate the `--audit-log-path` in `TestAuditLogs` but I'll do that as a follow up given it's currently working, and will depend on the ScratchDirs changes we've been discussing in #2407

**Pull request merged**
